[doc][core][dashboard][state] Promote state api and dashboard usage in core user guide. (#35760)
ray-project/ray: doc/source/ray-core/actors.rst

          // as the argument.
          auto counter = ray::Actor(CreateCounter).Remote();

> From Ray 2.0 onwards, you can use the :ref:`State API <state-api-overview-ref>`  to query for information about actors, or access the Actors View in the :ref:`Dashboard <observability-getting-started>` 

Reviewer: Actually it seems more natural to make this part simple. Let's just do this?

"Use `ray list tasks <that has a link to the CLI>` to see a list of tasks that run". 

```
 # This API is only available when you install Ray with `pip install "ray[default]"`.
ray list tasks
```

```
  ======== List: 2023-05-25 10:10:50.095099 ========
  Stats:
  ------------------------------
  Total: 1
  
  Table:
  ------------------------------
      ACTOR_ID                          CLASS_NAME    STATE      JOB_ID  NAME    NODE_ID                                                     PID  RAY_NAMESPACE
   0  9e783840250840f87328c9f201000000  Counter       ALIVE    01000000          13a475571662b784b4522847692893a823c78f1d3fd8fd32a2624923  38906  ef9de910-64fb-4575-8eb5-50573faa3ddf
```

          // as the argument.
          auto counter = ray::Actor(CreateCounter).Remote();

.. tip::
    

>     From Ray 2.0 onwards, you could also use :ref:`Ray's State API <state-api-overview-ref>``  to query the actors information, or go to the :ref:`Dashboard <observability-getting-started>`` Actor page to see more details. (These features all require installing ray with `pip install "ray[default]"`). 

Reviewer 2: ```suggestion
    From Ray 2.0 onwards, you can also use the :ref:`Ray's State API <state-api-overview-ref>``  to query for information about actors, or access the Actors View in the :ref:`Dashboard <observability-getting-started>`` to see more details. (These features require installing Ray with `pip install "ray[default]"`). 
```

>     From Ray 2.0 onwards, you could also use :ref:`Ray's State API <state-api-overview-ref>``  to query the actors information, or go to the :ref:`Dashboard <observability-getting-started>`` Actor page to see more details. (These features all require installing ray with `pip install "ray[default]"`). 

Reviewer: Let's just say "you can see if actors are created using ray list actors or the dashboard (link)" -> and have a short note saying it is only available for 2.5. Something similar to pg doc

<img width="954" alt="Screen Shot 2023-05-25 at 3 10 48 PM" src="https://github.com/ray-project/ray/assets/18510752/7d6c95d9-7501-43fe-99e0-c16ef4ba4077">

.. code-block:: bash

>   # This API is only available when you download Ray via `pip install "ray[default]"`.

Reviewer 2: ```suggestion
  # This API is only available when you install Ray with `pip install "ray[default]"`.
```
